src/internal.c

```diff
@@ -6803,9 +6803,39 @@ int SetSSL_CTX(WOLFSSL* ssl, WOLFSSL_CTX* ctx, int writeDup)
 #endif /* HAVE_RPK */
 
 #ifndef NO_CERTS
+#ifdef WOLFSSL_COPY_CERT
+    /* If WOLFSSL_COPY_CERT is defined, always copy the cert */
+    if (ctx->certificate != NULL) {
+        if (ssl->buffers.certificate != NULL) {
+            FreeDer(&ssl->buffers.certificate);
+        }
+        ret = AllocCopyDer(&ssl->buffers.certificate, ctx->certificate->buffer,
```

Comment:
Don't you need to set the `ssl->buffers.weOwnCert` and `weOwnCertChain` to 1?

Reply:
So it seemed to me I had two options when coding this, try to hook into the existing weOwnCert logic or do separate overrides independent of the weOwnCert logic.  I chose the latter, I thought it was simpler to do it that way but I didnt feel strongly either way. If you want me to rework this to use weOwnCert I can take a look at it.

Reply:
I think you'll need to. What happens if you make a copy then try and load another cert on the WOLFSSL object? I think it will leak it...

Reply:
Well luckily the unit test actually covers a variety of use cases including that one. I am fairly certain that case is covered and doesnt leak. That being said though I am definitely not opposed to using weOwnCert. I was on the fence about it already, so if you think its a good idea I can take a look at it.